src/sql/src/session/vars/value.rs

```diff
@@ -479,6 +480,44 @@ impl Value for Vec<Ident> {
     }
 }
 
+impl Value for Vec<IpNet> {
+    fn type_name() -> Cow<'static, str>
+    where
+        Self: Sized,
+    {
+        "CIDR list".into()
+    }
+
+    fn parse(input: VarInput<'_>) -> Result<Self, VarParseError>
+    where
+        Self: Sized,
+    {
+        // This matches VarInput.to_vec, but will split the flat input on ','
+        let values: Vec<String> = match input {
+            VarInput::SqlSet(values) => values.into_iter().map(|v| v.to_string()).collect(),
+            VarInput::Flat(value) => value.split(",").map(|s| s.trim().to_owned()).collect(),
+        };
```

Comment:
Is there a particular reason why we want to split the flat input on a `,`? The [parser](https://github.com/MaterializeInc/materialize/blob/main/src/sql-parser/src/parser.rs#L7159-L7167) should already be splitting multiple values by a comma. If it's not then we should fix that :)

Reply:
Yeah, this was honestly the hardest thing to figure out.. the parser does do it right, but it is unfortunately being turned back into a `VarInput::Flat` later, which will cause a panic. I'm actually guessing this would cause a panic in any system var that takes a list... I wasn't sure so I just fixed it here.


https://github.com/MaterializeInc/materialize/blob/main/src/adapter/src/config/params.rs#L115

Reply:
Darn you're totally right, we need to do an audit of where `VarInput::Flat` is used and probably refactor to thread through the concept of multiple values. Definitely not a block for this PR!

Reply:
Filed https://github.com/MaterializeInc/database-issues/issues/8595 to track